model/windowInformation.h

```diff
@@ -0,0 +1,63 @@
+/*
+  @copyright Steve Keen 2021
+  @author Janak Porwal
+  This file is part of Minsky.
+
+  Minsky is free software: you can redistribute it and/or modify it
+  under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Minsky is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Minsky.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef WINDOW_INFORMATION_H
+#define WINDOW_INFORMATION_H
+
+#include <cairoSurfaceImage.h>
+#include <X11/Xlib.h>
+
+namespace minsky
+{
+  class WindowInformation
+  {
+    unsigned long parentWindowId;
+    unsigned long childWindowId;
+    
+    Display*	display; // Weak reference, returned by system
+    ecolab::cairo::SurfacePtr childSurface;
+
+  public: 
+      int childWidth;
+      int childHeight;
+      int offsetLeft;
+      int offsetTop;
+      
+      XWindowAttributes wAttr;
+      unsigned long getChildWindowId();
+      Display* getDisplay();
+
+  public:
+
+    
+    WindowInformation();
+    ~WindowInformation();
+    WindowInformation(unsigned long parentWin, int left, int top, int cWidth, int cHeight);
+    void initialize(unsigned long parentWin, int left, int top, int cWidth, int cHeight);
+    void initialize(const WindowInformation &other);
+
+    ecolab::cairo::SurfacePtr getSurface();
+
+    void copyFrom(const WindowInformation &other);
```

Comment:
I don't see the implementation of this method here, but I'm concerned you're trying to introduce a copy operator in by the back door, and no correctly handling resource lifetime (like the initialize() example above).